cvat-core/src/axios-tus.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2024 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import Axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+import * as tus from 'tus-js-client';
+
+class AxiosHttpResponse implements tus.HttpResponse {
+    readonly #axiosResponse: AxiosResponse;
+
+    constructor(axiosResponse: AxiosResponse) {
+        this.#axiosResponse = axiosResponse;
+    }
+
+    getStatus(): number { return this.#axiosResponse.status; }
+    getHeader(header: string): string { return this.#axiosResponse.headers[header.toLowerCase()]; }
+    getBody(): string { return this.#axiosResponse.data; }
+    getUnderlyingObject(): any { return this.#axiosResponse; }
+}
+
+class AxiosHttpRequest implements tus.HttpRequest {
+    readonly #axiosConfig: AxiosRequestConfig;
+    readonly #abortController: AbortController;
+
+    constructor(method: string, url: string) {
+        this.#abortController = new AbortController();
+        this.#axiosConfig = {
+            method, url, headers: {}, signal: this.#abortController.signal,
+        };
+    }
+
+    getMethod(): string { return this.#axiosConfig.method; }
+    getURL(): string { return this.#axiosConfig.url; }
+
+    setHeader(header: string, value: string): void {
+        this.#axiosConfig.headers[header.toLowerCase()] = value;
+    }
+    getHeader(header: string): string | undefined {
+        return this.#axiosConfig.headers[header.toLowerCase()];
+    }
+
+    setProgressHandler(handler: (bytesSent: number) => void): void {
+        this.#axiosConfig.onUploadProgress = (progressEvent) => {
+            handler(progressEvent.loaded);
+        };
+    }
+
+    async send(body: any): Promise<tus.HttpResponse> {
+        const axiosResponse = await Axios({ ...this.#axiosConfig, data: body });
+        return new AxiosHttpResponse(axiosResponse);
+    }
+
+    async abort(): Promise<void> { this.#abortController.abort(); }
```

Comment:
```suggestion
    async abort(): Promise<void> {
        await this.#abortController.abort();
    }
```

Reply:
Just looked at the implementation of abort controller. It looks like the method is sync. Is there any reason to make it this wrapper method async here?

Reply:
The `abort` function is [defined](https://github.com/tus/tus-js-client/blob/main/docs/api.md#httpstack) as returning a promise (I don't really know why...), so I made it `async` to make JS return a promise implicitly.